crates/rspack_plugin_javascript/src/visitors/dependency/harmony_detection_scanner.rs

```diff
@@ -106,7 +115,13 @@ impl Visit for TopLevelAwaitScanner {
   fn visit_constructor(&mut self, _: &Constructor) {}
   fn visit_function(&mut self, _: &Function) {}
 
-  fn visit_await_expr(&mut self, _: &AwaitExpr) {
-    self.has_top_level_await = true;
+  fn visit_await_expr(&mut self, await_expr: &AwaitExpr) {
+    if self.top_level_await.is_none() {
+      self.top_level_await = Some(Span::new(
+        await_expr.span.span_lo(),
+        await_expr.arg.span_lo(),
```

Comment:
Let's use `await_expr.span.span_lo() + 5` as the end of this span?

Reply:
This will make it more precise:
```
  × Module parse failed:
  ╰─▶   × JavaScript parsing error: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)
         ╭─[1:1]
       1 │ const a = await Promise.resolve("aaa");
         ·           ──────
       2 │ export default a;
         ╰────
```